pkg/controllers/state/statenode.go

```diff
@@ -72,18 +72,19 @@ func (n StateNodes) Pods(ctx context.Context, kubeClient client.Client) ([]*v1.P
 	return pods, nil
 }
 
-// Disruptable filters StateNodes that are meet the IsDisruptable condition
-func (n StateNodes) Disruptable(ctx context.Context, clk clock.Clock, kubeClient client.Client) (StateNodes, error) {
-	pdbs, err := pdb.NewLimits(ctx, clk, kubeClient)
-	if err != nil {
-		return StateNodes{}, fmt.Errorf("constructing pdbs, %w", err)
-	}
-	n = lo.Filter(n, func(node *StateNode, _ int) bool {
-		_, err := node.ValidateDisruptable(ctx, kubeClient, pdbs)
-		return err == nil
-	})
-	return n, nil
-}
+// TODO: this doesn't seem to be called? is it needed?
+// // Disruptable filters StateNodes that are meet the IsDisruptable condition
+// func (n StateNodes) Disruptable(ctx context.Context, clk clock.Clock, kubeClient client.Client) (StateNodes, error) {
+// 	pdbs, err := pdb.NewLimits(ctx, clk, kubeClient)
+// 	if err != nil {
+// 		return StateNodes{}, fmt.Errorf("constructing pdbs, %w", err)
+// 	}
+// 	n = lo.Filter(n, func(node *StateNode, _ int) bool {
+// 		_, err := node.ValidateDisruptable(ctx, kubeClient, pdbs)
+// 		return err == nil
+// 	})
+// 	return n, nil
+// }
```

Comment:
@njtran What's the plan for this function? Right now it looks like it's unused, but I suspect you have a plan in mind for it? 

`ValidateDisruptible()` needs to consider a new `disruptionClass` string and the `terminationGracePeriod` of the associated node.

Reply:
I've opted to add both TGP and disruptionClass to `Disruptible()` and `ValidateDisruptible()`. Feel free to start a thread if you see any issues with this approach.